sdk/python/kfp/_credentials.py

```diff
@@ -42,3 +50,38 @@ def refresh_api_key_hook(self, config: Configuration):
     @abc.abstractmethod
     def get_token(self):
         raise NotImplementedError()
+
+
+def read_sa_token(path=None):
+    """Read a ServiceAccount token found under some path."""
+    token = None
+    with open(path, "r") as f:
+        token = f.read().strip()
+    return token
+
+
+class ServiceAccountTokenVolumeCredentials(TokenCredentialsBase):
+    """Audience-bound ServiceAccountToken in the local filesystem.
+
+    This is a credentials interface for audience-bound ServiceAccountTokens
+    found in the local filesystem, that get refreshed by the kubelet.
+
+    The constructor of the class expects a filesystem path.
+    If not provided, it uses the path stored in the environment variable
+    defined in ML_PIPELINE_SA_TOKEN_ENV.
+    If the environment variable is also empty, it falls back to the path
+    specified in ML_PIPELINE_SA_TOKEN_PATH.
+
+    This method of authentication is meant for use inside a Kubernetes cluster.
+
+    Relevant documentation:
+    https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection
+    """
+
+    def __init__(self, path=None):
+        self.token_path = (path
```

Comment:
can this be private? `self._token_path` ?

Reply:
Done!